haystack/utils/docstore_deserialization.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict
+
+from haystack import DeserializationError
+from haystack.core.serialization import default_from_dict, import_class_by_name
+
+
+def deserialize_document_store_in_init_parameters(data: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Deserializes a generic document store from the init_parameters of a serialized component.
+
+    :param data:
+        The dictionary to deserialize from.
+    :returns:
+        The dictionary, with the document store deserialized.
+
+    :raises DeserializationError:
+        If the document store is not properly specified in the serialization data or its type cannot be imported.
+    """
+    init_params = data.get("init_parameters", {})
+    if "document_store" not in init_params:
```

Comment:
Hmm, this assumption could be problematic for components that might have multiple, differently named init parameters that are document stores. On that note, I think we can make this function even more generic, i.e., we can call it `deserialize_component_or_document_store_in_init_parameters`, accept the key as an argument and deserialize the value of the particular key.

Reply:
I exposed the `key` argument